lib/OptimizationMOI/src/OptimizationMOI.jl

```diff
@@ -358,6 +358,16 @@ function SciMLBase.__solve(prob::OptimizationProblem,
             end
         end
     end
+
+    if prob.int !== nothing
+        @assert eachindex(prob.int) == Base.OneTo(num_variables)
+        for i in 1:num_variables
+            if prob.int[i]
+                MOI.add_constraint(opt_setup, θ[i], MOI.ZeroOne())
```

Comment:
The integer flag does not imply binary variables, only integrality. If additionally `(lb,ub)=(0,1)`, then it's binary.